# Cross check the result:

# Importing clean dataset:

In [1]:
import pandas as pd

In [2]:
dataset = pd.read_csv("https://raw.githubusercontent.com/DINESHGITAYE/Filter-Emails/main/dataset.csv")

In [3]:
dataset.head()

From  \
0                     linkedin    
1                     amod oke    
2          "sharekhan limited"    
3          "sharekhan limited"    
4  saurav thakkar via linkedin    

                                             Subject  
0  morgan stanley is looking for: industrial trai...  
1     dinesh, please add me to your linkedin network  
2  post-market report: banks, financials lift mar...  
3  b'invitation for an exclusive webinar \x96 ind...  
4  dinesh, start a conversation with your new con...

# Filtering out the Emails that are related to JOB:

In [4]:
dataset[dataset.apply(lambda row: row.astype(str).str.contains('job').any(), axis= 1)]

From  \
7                  thinkapps solutions private limited    
8                                        sai placement    
22                                       naukri alerts    
24     "gratitude india manpower consultants pvt. ltd."   
26                                       naukri alerts    
...                                                 ...   
990   "gratitude india manpower consultants pvt. ltd."    
991        emagine people technologies private limited    
998                                      naukri alerts    
1000                                     naukri alerts    
1001                    future vision career solutions    

                                                Subject  
7     job | urgent opening with a top it company - f...  
8     job | manufacturing chemist for mumbai tarapur...  
22    dinesh gitaye, check out the jobs applied by y...  
24    new job - data analyst / senior data analyst |...  
26    job | (k)your cv has been shortlisted for high...  
...                                                 ...  
990   job | covid-19: work from home in a top bpo in...  
991   job | immediate opportunity for phone recruite...  
998   new job - risk analyst | more jobs matching yo...  
1000  dinesh gitaye, check out the jobs applied by y...  
1001          job | urgent opening for andheri loaction  

[177 rows x 2 columns]

### > Now, see how many Emails are related to JOB:

In [5]:
Emails_related_to_job = dataset[dataset.apply(lambda row: row.astype(str).str.contains('job').any(), axis= 1)]
Emails_related_to_job['Subject'].count()

177

##### There are 177 emails that are related to job

# > To see all these 177 emails which is related to job:

In [6]:
pd.set_option('display.max_rows', Emails_related_to_job.shape[0]+1)
Emails_related_to_job

From  \
7                  thinkapps solutions private limited    
8                                        sai placement    
22                                       naukri alerts    
24     "gratitude india manpower consultants pvt. ltd."   
26                                       naukri alerts    
27                                       naukri alerts    
29                                       naukri alerts    
33                                   mehta enterprises    
35                                  akash vikram kadam    
39                            =?utf-8?b?tglua2vksw4=?=    
41                                       naukri alerts    
52                                       naukri alerts    
53                                       naukri alerts    
66                                       naukri alerts    
68                  naukri premium - employer services    
77                                       naukri alerts    
82                                       naukri alerts    
84                             infinx services pvt ltd    
93                                       naukri alerts    
106                                      naukri alerts    
130                                      naukri alerts    
136                            career zone consultancy    
138                                      naukri alerts    
140           "trivium education services (p) limited"    
156                                      naukri alerts    
170                                      naukri alerts    
194                                      naukri alerts    
201                                      naukri alerts    
212                                      naukri alerts    
218     inventurus knowledge solutions private limited    
226                                      naukri alerts    
228                                chegg india pvt ltd    
234                                      career divine    
238                                      naukri alerts    
249   marsh mclennan global services india private l...   
250                                      naukri alerts    
252                                      naukri alerts    
256                                            caliber    
258                                      naukri alerts    
263                                      naukri alerts    
277                                chegg india pvt ltd    
298                             i expert international    
299                             i expert international    
301                                         compliance    
302                                      naukri alerts    
307                                      naukri alerts    
313                                      naukri alerts    
323                                      naukri alerts    
324                                      naukri alerts    
331                                           linkedin    
344                                         compliance    
346                                      naukri alerts    
351                                      naukri alerts    
358                           =?utf-8?b?tglua2vksw4=?=    
361                                  "naukri.com info"    
366                                      naukri alerts    
374                                  "naukri.com info"    
384                             i expert international    
385                                      naukri alerts    
387                               surbhi adarsh ranjan    
389                                             naukri    
393                                         job circle    
407                                      naukri alerts    
413                                      naukri alerts    
417               shunya tattva management consultants    
420                                           linkedin    
423               shunya tattva management consultants    
424                             relianc

## Note:
- These mails are Filter out based on keyword **'job'** because I do not have any mail who have a subject like 'Thank you for applying'. But we can also do it for keyword 'Thank you for applying'.  

- for this, we have to change only one thing in line number **4**. In place of **"job"** we only have to put **"thank you for applying"**.

- That is we have to write code like:

  * dataset[dataset.apply(lambda row: row.astype(str).str.contains(**'thank you for applying'**).any(), axis= 1)]